plugins/generator-1.18.2/forge-1.18.2/templates/rangeditem/rangeditem.java.ftl

```diff
@@ -143,17 +147,20 @@ public class ${name}Item extends Item {
 			if (stack.hurt(1, world.getRandom(), entity)) {
 				stack.shrink(1);
 				stack.setDamageValue(0);
-            	if (stack.isEmpty())
-               		entity.getInventory().removeItem(stack);
+				if (stack.isEmpty())
+					entity.getInventory().removeItem(stack);
 			}
 		} else{
 			stack.shrink(1);
-            if (stack.isEmpty())
-               entity.getInventory().removeItem(stack);
+			if (stack.isEmpty())
+			   entity.getInventory().removeItem(stack);
 		}
 	}
 	<#else>
-	entityarrow.pickup = AbstractArrow.Pickup.DISALLOWED;
+	if (entity.getAbilities().instabuild)
+	entityarrow.pickup = AbstractArrow.Pickup.CREATIVE_ONLY;
+	else
+	entityarrow.pickup = AbstractArrow.Pickup.<#if !data.pickupProjectiles>DISALLOWED<#else>ALLOWED</#if>;

```

Comment:
Apply this to the 1.19 template too. This way the else branch is generated only if the pickup option is disabled

```suggestion
	if (entity.getAbilities().instabuild)
		entityarrow.pickup = AbstractArrow.Pickup.CREATIVE_ONLY;
	<#if !data.pickupProjectiles>
	else
		entityarrow.pickup = AbstractArrow.Pickup.DISALLOWED;
	</#if>
```